### PR TITLE
RFC Ed version

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

File: rfc8826.xml [hunk 1]
```diff
@@ -22,60 +24,74 @@
...
+document (but not the title) as Web Real-Time Communication.  Do you want to
+include this expansion somewhere, or is not needed with the current
+explanatory text? 
+
+Original (first occurrence):
+   WebRTC is a protocol suite for use with real-time applications that
+   can be deployed in browsers - "real time communication on the Web".
+   This document defines the WebRTC threat model and analyzes the
+   security threats of WebRTC in that model.
+-->
```
Comment (ekr) [inline rfc8826.xml:49]: I don't think this is necessary..

File: rfc8826.xml [hunk 2]
```diff
@@ -22,60 +24,74 @@
...
+                |           |    Media   |           |
+                |  Browser  |<---------->|  Browser  |
+                |           |            |           |
+                +-----------+            +-----------+
+                    Alice                     Bob ]]></artwork>
       </figure>
       <t>
         In the system shown in <xref target="fig.simple" format="default"/>, Alice and Bob both have
-        WebRTC-enabled browsers and they visit some Web server which operates a
+        WebRTC-enabled browsers and they visit some Web server that operates a
```
Comment (ekr) [inline rfc8826.xml]: Please revert all that/which changes. See: 
http://itre.cis.upenn.edu/%7Emyl/languagelog/archives/001464.html

File: rfc8826.xml [hunk 3]
```diff
@@ -89,19 +105,19 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+        security challenges beyond those of a conventional Voice over IP (VoIP) system. In particular,
         it needs to contend with malicious calling services.
         For example, if the calling service
         can cause the browser to make a call at any time to any callee of its
         choice, then this facility can be used to bug a user's computer without
         their knowledge, simply by placing a call to some recording service.
         More subtly, if the exposed APIs allow the server to instruct the
         browser to send arbitrary content, then they can be used to bypass
-        firewalls or mount denial of service attacks. Any successful system
+        firewalls or mount DoS attacks. Any successful system
```
Comment (ekr) [inline rfc8826.xml]: please expand DoS on first use.

File: rfc8826.xml [hunk 4]
```diff
@@ -143,16 +160,26 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
       </t>
       <t>
-        Conventionally, we refer to either web attackers, who are able to induce
+        Conventionally, we refer to either Web attackers, who are able to induce
         you to visit their sites but do not control the network, and network
-        attackers, who are able to control your network. Network attackers correspond
+        attackers, who are able to control your network.
+
+<!-- [rfced] Section 3:  Should "network, and" be "network, or," or
+should the word "either" be removed?
```
Comment (ekr) [inline rfc8826.xml:177]: Either/or.

File: rfc8826.xml [hunk 5]
```diff
@@ -188,31 +215,60 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
-          DoS attack).
+          (e.g., from misuse of his credentials) and server B (e.g., from
+          DoS attacks).
+
+<!-- [rfced] Section 3.2:  Are "ORIGIN" and "SAME ORIGIN POLICY"
+written in all capitals for emphasis (in which case perhaps we
+could use the <strong> element (Section 2.50 of RFC 7991)), or should
+we write them as "origin" (as used elsewhere in this document and
+this cluster) and "Same-Origin Policy" (as used elsewhere in this
+document and in several published RFCs)?
```
Comment (ekr) [inline rfc8826.xml:239]: Origin should be emphasized. SAME-ORIGIN POLICY need not be.

File: rfc8826.xml [hunk 6]
```diff
@@ -188,31 +215,60 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+
+Also, per the "Gender-Specific Language" section of
+<https://www.rfc-editor.org/styleguide/part2/>, please let us know
+if we may change instances of "his," "him," "himself," and "he" to
+"their," "them," "themselves," and "they."
```
Comment (ekr) [inline rfc8826.xml:246]: Yes.

File: rfc8826.xml [hunk 7]
```diff
@@ -188,31 +215,60 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+
         </t>
         <t>
           More generally, SOP forces scripts from each site to run in their own, isolated,
           sandboxes. While there are techniques to allow them to interact, those interactions
           generally must be mutually consensual (by each site) and are limited to certain
-          channels. For instance, multiple pages/browser panes from the same origin
-          can read each other's JS variables, but pages from the different origins--or
-          even iframes from different origins on the same page--cannot.
+          channels. For instance, multiple pages / browser panes from the same origin
```
Comment (ekr) [inline rfc8826.xml]: Revert the spaces here.

File: rfc8826.xml [hunk 8]
```diff
@@ -188,31 +215,60 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+          can read each other's JS variables, but pages from different
+	  origins -- or
+          even iframes from different origins on the same page -- cannot.
         </t>
-        <!-- TODO: Picture -->
+
+<!-- [rfced] Section 3.3:  We found a "TODO: Picture" comment in the
+XML file, just after the section title.  Is a diagram missing? -->
```
Comment (ekr) [inline rfc8826.xml]: I think I must have just decided not to.

File: rfc8826.xml [hunk 9]
```diff
@@ -258,22 +314,22 @@ the title) for use on https://www.rfc-editor.org/search. -->
           In order for the user to
           make an intelligent decision about whether to allow a call
           (and hence his camera and microphone input to be routed somewhere),
-          he must understand either who is requesting access, where the media
+          he must understand who is requesting access, where the media
```
Comment (ekr) [inline rfc8826.xml]: Please revert either here.

File: rfc8826.xml [hunk 10]
```diff
@@ -291,7 +347,7 @@ the title) for use on https://www.rfc-editor.org/search. -->
           user's browser from being used to attack its local network.
           Thus, we need to ensure communications consent even if the
           site is not able to access the camera and microphone at
-          all (hence WebSockets's consent mechanism) and similarly
+          all (hence WebSockets's consent mechanism); similarly,
```
Comment (ekr) [inline rfc8826.xml]: Please revert this change.

File: rfc8826.xml [hunk 11]
```diff
@@ -310,7 +366,7 @@ the title) for use on https://www.rfc-editor.org/search. -->
           </t>
           <t>
             The most obvious threats are simply those of "oversharing".
-            I.e., the user may believe they are sharing a window when
+            That is, the user may believe they are sharing a window when
```
Comment (ekr) [inline rfc8826.xml]: Revert all I.e. -> That is.

File: rfc8826.xml [hunk 12]
```diff
@@ -415,18 +471,18 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+          but because the Web model is scoped to the origin, this creates a difficult
           mismatch.
           </t>
           <t>
-          Arguably, origin is not fine-grained enough. Consider the situation where
+          Arguably, the origin is not fine-grained enough. Consider the situation where
           Alice visits a site and authorizes it to make a single call. If consent is
-          expressed solely in terms of origin, then at any future visit to that
-          site (including one induced via mash-up or ad network), the site can
+          expressed solely in terms of the origin, then upon any future visit to that
```
Comment (ekr) [inline rfc8826.xml]: revert "upon"

File: rfc8826.xml [hunk 13]
```diff
@@ -457,52 +513,52 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+          describes mechanisms which facilitate this sort of consent.
           </t>
           <t>
           Another case where media-level cryptographic identity makes sense is when a user
           really does not trust the calling site. For instance, I might be worried that
           the calling service will attempt to bug my computer, but I also want to be
           able to conveniently call my friends. If consent is tied to particular
           communications endpoints, then my risk is limited. Naturally, it
-          is somewhat challenging to design UI primitives which express this sort
+          is somewhat challenging to design UI primitives that express this sort
```
Comment (ekr) [inline rfc8826.xml]: revert.

File: rfc8826.xml [hunk 14]
```diff
@@ -512,8 +568,19 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
           <xref target="finer-grained" format="default"/>. This could lead to compromise of a call
-          even if the parent page is safe. Note: this issue is not restricted
-          to PAGES which contain untrusted content. If any page from a
+          even if the parent page is safe. Note: This issue is not restricted
+          to PAGES which contain untrusted content.
+
+<!-- [rfced] Section 4.1.4:  Is "PAGES" capped for emphasis, or
+should it be "pages"?  (We see "a page" and "the page" used in nearby
+text in this section.)  If emphasis is desired, perhaps we could use
+the <strong> element (Section 2.50 of RFC 7991) here.
```
Comment (ekr) [inline rfc8826.xml:594]: Emphasis.

File: rfc8826.xml [hunk 15]
```diff
@@ -523,11 +590,12 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
         <name>Communications Consent Verification</name>
         <t>
-          As discussed in <xref target="sec.cors-etc" format="default"/>, allowing web applications unrestricted network access
+          As discussed in <xref target="sec.cors-etc" format="default"/>, allowing Web applications unrestricted network access
           via the browser introduces the risk of using the browser as an attack platform against
-          machines which would not otherwise be accessible to the malicious site, for
-          instance because they are topologically restricted (e.g., behind a firewall or NAT).
-          In order to prevent this form of attack as well as cross-protocol attacks it is
+          machines which would not otherwise be accessible to the malicious
+	  site -- for
```
Comment (ekr) [inline rfc8826.xml]: revert to comma

File: rfc8826.xml [hunk 16]
```diff
@@ -665,8 +735,23 @@ the title) for use on https://www.rfc-editor.org/search. -->
           <name>IP Location Privacy</name>
           <t>
             Note that as soon as the callee sends their ICE candidates, the
-            caller learns the callee's IP addresses. The callee's server reflexive
+            caller learns the callee's IP addresses. The callee's server-reflexive
             address reveals a lot of information about the callee's location.
+
+<!-- [rfced] Section 4.2.4:  Per author feedback for RFC 8839 and per
+other documents in this cluster, we hyphenated the term "server
+reflexive".  Please let us know any objections.
```
Comment (ekr) [inline rfc8826.xml:762]: Fine

File: rfc8826.xml [hunk 17]
```diff
@@ -710,10 +795,10 @@ the title) for use on https://www.rfc-editor.org/search. -->
           While in principle it is possible for the browser to cut the calling service
           out of the loop and directly present trusted information (and perhaps get
           consent), practice in modern browsers is to avoid this whenever possible.
-          "In-flow" modal dialogs which require the user to consent to specific
+          "In&nbhy;flow" modal dialogs which
```
Comment (ekr) [inline rfc8826.xml]: Uh,... This text got lost.

File: rfc8826.xml [hunk 18]
```diff
@@ -744,36 +829,54 @@ the title) for use on https://www.rfc-editor.org/search. -->
           and why they are likely to be impractical in <xref target="sec.during-attack" format="default"/>.
         </t>
         <section anchor="sec.retrospective-compromise" numbered="true" toc="default">
-          <name>Protecting Against Retrospective Compromise</name>
+          <name>Protecting against Retrospective Compromise</name>
```
Comment (ekr) [inline rfc8826.xml]: This seems like it should be capitalized.

File: rfc8826.xml [hunk 19]
```diff
@@ -744,36 +829,54 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
           </t>
           <t>
             If the calling service has access to the traffic keying material
             (as in SDES <xref target="RFC4568" format="default"/>), then retrospective attack
             is trivial.
-            This form of attack is particularly serious in the Web context because
+
+<!-- [rfced] Section 4.3.1:  We would like to expand "SDES" for ease
+of the reader.  Does SDES refer to "Security Description" here, or perhaps "Source Description" per
+several other documents in this cluster (e.g., RFC-to-be 8852 <draft-ietf-avtext-rid>)?
```
Comment (ekr) [inline rfc8826.xml]: Fine.

File: rfc8826.xml [hunk 20]
```diff
@@ -744,36 +829,54 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+
+<!-- [rfced] Section 4.3.1:  We would like to expand "SDES" for ease
+of the reader.  Does SDES refer to "Security Description" here, or perhaps "Source Description" per
+several other documents in this cluster (e.g., RFC-to-be 8852 <draft-ietf-avtext-rid>)?  
+
+Original:
+ If the calling service has access to the traffic keying material (as
+ in SDES [RFC4568]), then retrospective attack is trivial. -->
+
+            This form of attack is particularly serious in the context of the Web because
```
Comment (ekr) [inline rfc8826.xml]: revert. "Web context.

File: rfc8826.xml [hunk 21]
```diff
@@ -744,36 +829,54 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
             it is standard practice in Web services to run extensive logging and monitoring. Thus, it is highly
             likely that if the traffic key is part of any HTTP request it will be logged somewhere and thus
             subject to subsequent compromise. It is this consideration that makes an automatic, public key-based
             key exchange mechanism imperative for WebRTC (this is a good idea for any communications
-            security system) and this mechanism <bcp14>SHOULD</bcp14> provide perfect forward secrecy (PFS).
-            The signaling channel/calling service can be used to authenticate this mechanism.
+            security system), and this mechanism <bcp14>SHOULD</bcp14> provide Perfect Forward Secrecy (PFS).
```
Comment (ekr) [inline rfc8826.xml]: Actually, we want to just change this to forward secrecy, as in TLS (check).

File: rfc8826.xml [hunk 22]
```diff
@@ -744,36 +829,54 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+            The signaling channel / calling service can be used to authenticate this mechanism.
```
Comment (ekr) [inline rfc8826.xml]: Fix spaces.

File: rfc8826.xml [hunk 23]
```diff
@@ -744,36 +829,54 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
           </t>
           <t>
-            In addition, if end-to-end keying is in used,
+            In addition, if end-to-end keying is used,
             the system <bcp14>MUST NOT</bcp14> provide any APIs to extract either long-term
             keying material or to directly access any stored traffic keys.
+
+<!-- [rfced] Section 4.3.1:  To what does "either" refer in this
+sentence?
```
Comment (ekr) [inline rfc8826.xml:891]: extract -> either extract.

File: rfc8826.xml [hunk 24]
```diff
@@ -744,36 +829,54 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+ directly access any stored traffic keys. -->
+
             Otherwise, an attacker who subsequently compromised the calling service
             might be able to use those APIs to recover the traffic keys and thus
             compromise the traffic.
           </t>
         </section>
         <section anchor="sec.during-attack" numbered="true" toc="default">
-          <name>Protecting Against During-Call Attack</name>
+          <name>Protecting against During-Call Attack</name>
```
Comment (ekr) [inline rfc8826.xml]: Capitals again here.

File: rfc8826.xml [hunk 25]
```diff
@@ -834,32 +937,53 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
             <t>
-              ZRTP <xref target="RFC6189" format="default"/> uses a "short authentication string" (SAS) which is derived
-              from the key agreement protocol. This SAS is designed to be compared
+              ZRTP <xref target="RFC6189" format="default"/> uses a "Short Authentication String" (SAS) which is derived
+              from the key agreement protocol.
+
+<!-- [rfced] Section 4.3.2.2:  Is the SAS always derived from the key
+agreement protocol (in which case "(SAS), which is" would be correct)
+or only sometimes derived from the key agreement protocol (in which
+case "(SAS) that is" would be correct)?
```
Comment (ekr) [inline rfc8826.xml:974]: This is correct as is. One could imagine a short authentication string which was not derived from the AKE, one supposes.

File: rfc8826.xml [hunk 26]
```diff
@@ -834,32 +937,53 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
               voice from one speaker to make it sound like another,
               are an active area of research.
               These systems are already good enough to fool both
-              automatic recognition systems <xref target="farus-conversion" format="default"/> and
+              automatic recognition systems <xref target="farrus-conversion" format="default"/> and
               humans <xref target="kain-conversion" format="default"/> in many cases, and are of course likely
               to improve in future, especially in an environment where the user just wants
               to get on with the phone call.
-              Thus, even if SAS is effective today, it is likely not to be so for much longer.
+              Thus, even if the SAS is effective today, it is likely not to be so for much longer.
```
Comment (ekr) [inline rfc8826.xml:1012]: This actually needs a rewrite in light of deep fakes.

File: rfc8826.xml [hunk 27]
```diff
@@ -873,22 +997,22 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
               The conventional approach to providing communications identity
-              has of course been to have some third party identity system
+              has of course been to have some third-party identity system
               (e.g., PKI) to authenticate the endpoints. Such mechanisms
               have proven to be too cumbersome for use by typical users
               (and nearly too cumbersome for administrators).
               However,
               a new generation of Web-based identity providers (BrowserID, Federated Google Login,
-              Facebook Connect, OAuth <xref target="RFC6749" format="default"/>, OpenID <xref target="OpenID" format="default"/>, WebFinger <xref target="RFC7033" format="default"/>), has recently been developed
+              Facebook Connect, OAuth <xref target="RFC6749" format="default"/>, OpenID <xref target="OpenID" format="default"/>, WebFinger <xref target="RFC7033" format="default"/>) has recently been developed
```
Comment (ekr) [inline rfc8826.xml]: Remove "recently"

File: rfc8826.xml [hunk 28]
```diff
@@ -931,9 +1055,9 @@ the title) for use on https://www.rfc-editor.org/search. -->
               that, it will be necessary to provide features whereby
               the site can verifiably give up access to the media streams.
               This verification must be possible both from the local
-              side and the remote side. I.e., users must be able to verify
+              side and the remote side. That is, users must be able to verify
```
Comment (ekr) [inline rfc8826.xml]: Revert.

File: rfc8826.xml [hunk 29]
```diff
@@ -960,9 +1084,32 @@ the title) for use on https://www.rfc-editor.org/search. -->
             the call and publish it on the Internet. Similarly, we do
             not attempt to prevent them from using voice or video processing
             technology from hiding or changing their appearance.
```
Comment (ekr) [inline rfc8826.xml]: to hide

File: rfc8826.xml [hunk 30]
```diff
@@ -1073,59 +1193,79 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+<http://rtc-web.alvestrand.com/home/papers/
+barth-security-prompt.pdf?attredirects=0> - steers to
+<https://672ad43e-a-6ea19bdf-s-sites.googlegroups.com/
+a/alvestrand.com/rtc-web/home/papers/
+barth-security-prompt.pdf?attachauth= ...(a very long string that is
+different each time)...&attredirects=0>.
+
+Is
+<http://rtc-web.alvestrand.com/home/papers/barth-security-prompt.pdf?attredirects=0>
+considered the most stable URL available?  Or, should the URL not be included
```
Comment (ekr) [inline rfc8826.xml:1297]: I'm fine with this URL

File: rfc8826.xml [hunk 31]
```diff
@@ -1155,31 +1294,59 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+Suggested:
+ [kain-conversion]
+            Kain, A. and M. Macon, "Design and Evaluation of a Voice
+            Conversion Algorithm based on Spectral Envelope Mapping
+            and Residual Prediction", Proceedings of the 2001 IEEE
+            International Conference on Acoustics, Speech, and Signal
+            Processing (ICASSP), DOI 10.1109/ICASSP.2001.941039, May
+            2001, <https://ieeexplore.ieee.org/document/941039>. -->
+
+        <reference anchor="farrus-conversion">
```
Comment (ekr) [inline rfc8826.xml]: I'm just going to fix this.

File: rfc8826.xml [hunk 32]
```diff
@@ -1214,70 +1380,133 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+<https://www.w3.org/TR/cors/>:
+
+[CORS]
+    Anne van Kesteren. Cross-Origin Resource Sharing. 2 June 2020. REC. URL:
+    https://www.w3.org/TR/cors/ 
+
+<https://www.w3.org/TR/2020/SPSD-cors-20200602/> says that new implementations
+should follow the "Fetch API Living Standard".
+
+Please review and let us know if any updates are needed.
```
Comment (ekr) [inline rfc8826.xml]: Yes, update please.

File: rfc8826.xml [hunk 33]
```diff
@@ -1214,70 +1380,133 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
-        <reference anchor="XmlHttpRequest" target="http://www.w3.org/TR/XMLHttpRequest/">
+<!-- [rfced] Informative References:  
+The URL provided for [SWF] in the original XML file -
+<http://www.adobe.com/content/dam/Adobe/en/devnet/swf/pdf/
+swf_file_format_spec_v10.pdf> - steers to
+<https://www.adobe.com/content/dam/acom/en/devnet/swf/pdf/
+swf_file_format_spec_v10.pdf>, which in turn yields a 404.
+Please provide a working and stable URL.
+
+Original:
```
Comment (ekr) [inline rfc8826.xml:1460]: TODO

File: rfc8826.xml [hunk 34]
```diff
@@ -1214,70 +1380,133 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+not, please provide the correct URL for the document listed below - perhaps 
+<https://www.w3.org/standards/history/fingerprinting-guidance> or
+<https://www.w3.org/TR/fingerprinting-guidance/>? 
+
+Original:
+ [Fingerprinting]
+            "Fingerprinting Guidance for Web Specification Authors
+            (Draft)", November 2013. -->
+
+        <reference anchor="OpenID" target="https://openid.net/specs/openid-connect-core-1_0.html">
```
Comment (ekr) [inline rfc8826.xml:1497]: TODO

File: rfc8826.xml [hunk 35]
```diff
@@ -1214,70 +1380,133 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+<https://www.w3.org/TR/2012/WD-XMLHttpRequest-20120117/>, which is
+partially obscured by a red box that says "This version is
+outdated!"  The link in the box in turn steers to the October 2016
+"XMLHttpRequest Level 1" page.
+
+Please advise.
+
+Original:
+ [XmlHttpRequest]
+            van Kesteren, A., "XMLHttpRequest Level 2", January 2012. -->
```
Comment (ekr) [inline rfc8826.xml]: Ask Anne

File: rfc8826.xml [hunk 36]
```diff
@@ -1295,10 +1524,35 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+            <contact fullname="Martin Thomson"/>, <contact fullname="Magnus Westerlund"/>.
+      </t>
+
+    </section>
   </back>
+
+<!-- [rfced] Please let us know how/if the following should be made
+consistent: 
+
+ iframe / IFRAME (also per draft-ietf-rtcweb-security-arch)
```
Comment (ekr) [inline rfc8826.xml]: IFRAME

File: rfc8826.xml [hunk 37]
```diff
@@ -1295,10 +1524,35 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+    </section>
   </back>
+
+<!-- [rfced] Please let us know how/if the following should be made
+consistent: 
+
+ iframe / IFRAME (also per draft-ietf-rtcweb-security-arch) 
+
+ calling-service.example.com/ vs. calling-service.example.com
+   (We also see "http://anything.example.org/".)
```
Comment (ekr) [inline rfc8826.xml:1543]: I don't think it matters.

File: rfc8826.xml [hunk 38]
```diff
@@ -1295,10 +1524,35 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+
+ CROSS-PROTOCOL attacks (Section 3.3) /
```
Comment (ekr) [inline rfc8826.xml:1548]: yes

File: rfc8826.xml (hunk 32 above)
Comment (ekr) [reply rfc8826.xml]: @annevk: can you help here?

File: rfc8826.xml (hunk 35 above)
Comment (ekr) [reply rfc8826.xml]: @annevk: can you provide a proposed Fetch reference.

File: rfc8826.xml (hunk 32 above)
Comment (annevk) [reply rfc8826.xml]: I think this should be as simple as replacing the CORS reference with a reference to Fetch (which includes a definition of the CORS protocol).

Following the format above, something like:

```
[FETCH]

    Anne van Kesteren. Fetch. Living Standard. URL:
    https://fetch.spec.whatwg.org/
```

File: rfc8826.xml (hunk 35 above)
Comment (annevk) [reply rfc8826.xml]: If you want to continue talking about XMLHttpRequest, this could be:

```[XMLHTTPREQUEST]

    Anne van Kesteren. XMLHttpRequest. Living Standard. URL:
    https://xhr.spec.whatwg.org/
```
But you could also replace the XMLHttpRequest examples with `fetch()` examples and use the Fetch reference I suggested above.

File: rfc8826.xml [hunk 39]
```diff
@@ -188,31 +215,60 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
+Also, per the "Gender-Specific Language" section of
+<https://www.rfc-editor.org/styleguide/part2/>, please let us know
+if we may change instances of "his," "him," "himself," and "he" to
+"their," "them," "themselves," and "they."
+
+Original "(as described below.)" has been fixed):
+ For instance,
+ while scripts are allowed to make HTTP requests via the
+ XMLHttpRequest() API (see [XmlHttpRequest]) those requests are not
+ allowed to be made to any server, but rather solely to the same
```
Comment (annevk) [inline rfc8826.xml:252]: So this is not factually correct. There are a number of requests that can go to any server and that roughly matches the capabilities of HTML form elements. This can either be considered part of the same-origin policy or enshrined exceptions to it.

I'm not sure how important those details are for this document however as WebRTC would not use those type of requests anyway (at which point CORS preflights or a TLS handshake become important).

Comment (ekr) [reply rfc8826.xml:252]: Yeah, not sure what I was thinking when I wrote that. Do you have a suggested replacement one-liner?

Comment (annevk) [reply rfc8826.xml:252]: Maybe something like:

> Scripts can make HTTP requests to other servers roughly within the confines of the functionality of HTML forms, but are not able to see responses of those requests unless the server cooperates via CORS. HTTP requests that go beyond HTML forms are preceded by a CORS preflight to ensure the server is prepared to handle them.

(Looking at https://datatracker.ietf.org/doc/draft-ietf-rtcweb-security/?include_text=1 it seems that would require changes in 3.2 and 3.3. You could also look at https://tools.ietf.org/html/rfc6454#section-3.4.2 for inspiration.)

Comment (ekr) [reply rfc8826.xml:252]: @annevk: I tried to shorten this some. PTAL.

File: rfc8826.xml [hunk 40]
```diff
@@ -184,36 +218,75 @@ the title) for use on https://www.rfc-editor.org/search. -->
...
-          while scripts are allowed to make HTTP requests via the XMLHttpRequest() API (see <xref target="XmlHttpRequest" format="default"/>)
-          those requests are not allowed to be made to any server, but rather solely
-          to the same ORIGIN from whence the script came <xref target="RFC6454" format="default"/>
-          (although CORS <xref target="CORS" format="default"/> and WebSockets
+          Many other resources are accessible but isolated.
+          For instance,
+          while scripts are allowed to make HTTP requests via the fetch() API (see <xref target="fetch" format="default"/>)
+          when requests are made to a server other than from the
+          the same <strong>origin</strong> from whence the script came <xref target="RFC6454" format="default"/>
+          they not able to read the responses. Cross-Origin Resource Sharing (CORS) <xref target="CORS" format="default"/> and WebSockets
```
Comment (annevk) [inline rfc8826.xml]:
```suggestion
          they are not able to read the responses. Cross-Origin Resource Sharing (CORS) <xref target="CORS" format="default"/> and WebSockets
```

Comment (annevk) [inline rfc8826.xml]: You probably still need to update the CORS reference here to point to Fetch instead.

**Review by annevk — COMMENTED** [review]
Looks good modulo nit.
